sdk/lusid/utilities/api_client_builder.py

```diff
@@ -10,15 +10,19 @@
 class ApiClientBuilder:
 
     @staticmethod
-    def build(api_secrets_filename=None, okta_response_handler=None):
+    def build(api_secrets_filename=None, okta_response_handler=None, api_configuration=None):
         """
         :param api_secrets_filename: name api configuration file
+        :param api_configuration: populated ApiConfiguration, if supplied this is used in preference to api_secrets_filename
         :param okta_response_handler: optional function to handle Okta response
         :return: ApiClient correctly configured with credentials and host
         """
 
+        if api_secrets_filename is None and api_configuration is None:
+            raise ValueError("Missing configuration, values must be supplied via api_configuration or a file path via api_secrets_filename")
+
```

Comment:
Shouldn't this fall back to environment variables in the case that neither of these are specified?

Reply:
good point, have removed